### PR TITLE
feat:cart management

### PR DESCRIPTION
**Pull request by teja538**
- did cart management
- add to cart button changes to go to cart on click
- did add to cart functionality
- did remove from cart functionality
- quantity increase and corresponding price changes
- preview link : https://deploy-preview-4--meek-longma-848475.netlify.app/

### CODE REVIEW

File: src/components/nav.jsx [hunk 1]
```diff
@@ -19,7 +22,7 @@ const Nav = () =>
             <li class="nav-item list-item-inline"><Link to="/cart" className="no-decoration color-black-light">
                     <span className="badge-on-icon">
                         <i className="fa fa-shopping-cart icon icon-outline"></i>
-                        <span className="icon-badge">6</span>
+                        <span className="icon-badge">{cartItems}</span>
```
Comment (Chiragmodi01) [inline src/components/nav.jsx:25]: Badge in the ui looks egg shaped than round.

File: src/functions/cart-reducer.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+const cartReducerFunc = (state, action) => {
+    switch (action.type) {
+      case "ADD-TO-CART":
+        return {
+          ...state,
+          cartItems: state.cartItems + 1,
+          cartList: [...state.cartList, {...action.value,cartQuantity:1}],
+          price:state.price + action.value.originalPrice,
+          discount: state.discount + (action.value.originalPrice - action.value.price),
+          totalPrice: state.totalPrice + action.value.price,
```
Comment (Chiragmodi01) [inline src/functions/cart-reducer.js:11]: It seems you're using these function many times below as well with just value changes, so you can make a separate function and pass value to change.
Looks clustered now.

File: src/pages/cart/cart.css [hunk 3]
```diff
@@ -0,0 +1,75 @@
+.head-center{
+    text-align: center;
+    padding: 2rem;
+}
+
+#cart-page
```
Comment (Chiragmodi01) [inline src/pages/cart/cart.css]: Only if styles are not applying with classes then use id.

File: src/pages/cart/cart.css [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+{
+    display: grid;
+    width: auto;
+    margin:0% 9%;
+    grid-row-gap: 2.5rem;
+    grid-template-areas: "cart-items price-details";
```
Comment (Chiragmodi01) [inline src/pages/cart/cart.css:12]: if you've only two areas prefer avoiding template and manage individual.

File: src/pages/cart/cart.jsx [hunk 5]
```diff
@@ -1,8 +1,77 @@
...
+                    <button className="btn filled-gold-btn"><Link to="/productlist" className="no-decoration color-white">Start Shopping</Link></button>
+                </div> 
+                :
+                <div id="cart-page">
+                    <div className="cart-items">
+                        {   
+                            cartList.map((pro)=>(
+                                <div className="card-horizontal cart-card" key={pro._id}>
+                                    <img className="card-img" src={pro.image} alt="card_image" />
+                                    <div className="card-info">
```
Comment (Chiragmodi01) [inline src/pages/cart/cart.jsx]: You can beautify a bit, code looks messy.

File: src/pages/product-listing/product-listing.css [hunk 6]
```diff
@@ -1,14 +1,21 @@
...
+    border: 1.8px solid var(--gold-dark);
+}
+.filled-gold-btn:hover
 {
-    background-color: var(--gold-color);
-    border-radius: 0px;
-    color:var(--white);
-    width: 100%; 
-    cursor: pointer;
+    background-color: var(--gold-dark);
```
Comment (Chiragmodi01) [inline src/pages/product-listing/product-listing.css:13]: good that you've used color variables at all places.

File: src/pages/product-listing/product-listing.jsx [hunk 7]
```diff
@@ -1,18 +1,22 @@
...
     const {state} = useProductListing()
     const {dispatch} = useProductListing()
+    const {cartDispatch,cartState} = useCart()
     const categoryFilterProducts = getFilterByCategory(allProducts,state.categoryFilterBy.rings,state.categoryFilterBy.bangles,state.categoryFilterBy.necklace,
         state.categoryFilterBy.earrings,state.categoryFilterBy.pendants)
     const ratingFilteredProducts = getFilterByRating(categoryFilterProducts,state.rating)
     const priceFilteredProducts = getFilterByPrice(ratingFilteredProducts,state.price)
     const finalProducts = getSortBy(priceFilteredProducts,state.sortBy)
-    console.log(finalProducts)
+    const cartList = cartState.cartList
```
Comment (Chiragmodi01) [inline src/pages/product-listing/product-listing.jsx]: see how clustered it looks. beautify a bit and if needed destructure values.

File: src/pages/product-listing/product-listing.jsx [hunk 8]
```diff
@@ -1,18 +1,22 @@
...
     return(
         <div className="prod-page">
             <div className="aside">
```
Comment (Chiragmodi01) [inline src/pages/product-listing/product-listing.jsx]: aside tag can be used here semantically

**Review by Chiragmodi01 — COMMENTED** [review]
- Move to wishlist feature dosen't seem to work.
Overall code structure and UI looks good just added few sanity comments.

**Review by Balakumar3018 — APPROVED** [review]
UI looks clean and Good to merge.

**Comment by teja538** [comment]
> * Move to wishlist feature dosen't seem to work.
>   Overall code structure and UI looks good just added few sanity comments.

Thank you for all the suggestions chirag! will make changes accordingly

**Pull request merged**